### PR TITLE
Add getBlockState() to PlayerInteractEvent

### PR DESCRIPTION
**Pull request by Inskey**
Added a `.getBlockState()` method to `PlayerInteractEvent` for clarity and ease of use. Returns null if the event is not a block interaction event, based on the value of `face`.

### CODE REVIEW

File: patches/minecraft/net/minecraft/world/ChunkCache.java.patch [hunk 1]
```diff
@@ -46,7 +46,7 @@
 +    {
 +        int x = (pos.func_177958_n() >> 4) - this.field_72818_a;
 +        int z = (pos.func_177952_p() >> 4) - this.field_72816_b;
-+        if (pos.func_177956_o() >= 0 && pos.func_177956_o() < 256) return _default;
++        if (pos.func_177956_o() < 0 && pos.func_177956_o() >= 256) return _default;
 +        if (x < 0 || x >= field_72817_c.length || z < 0 || z >= field_72817_c[x].length) return _default;
 +        if (field_72817_c[x][z] == null) return _default;
 +
```
Comment (williewillus) [inline patches/minecraft/net/minecraft/world/ChunkCache.java.patch]: what is this patch doing here?

File: patches/minecraft/net/minecraft/world/WorldEntitySpawner.java.patch [hunk 2]
```diff
@@ -28,7 +28,7 @@
 +                                                net.minecraftforge.fml.common.eventhandler.Event.Result canSpawn = net.minecraftforge.event.ForgeEventFactory.canEntitySpawn(entityliving, p_77192_1_, f, i3, f1);
 +                                                if (canSpawn == net.minecraftforge.fml.common.eventhandler.Event.Result.ALLOW || (canSpawn == net.minecraftforge.fml.common.eventhandler.Event.Result.DEFAULT && (entityliving.func_70601_bi() && entityliving.func_70058_J())))
                                                  {
-+                                                    if (!net.minecraftforge.event.ForgeEventFactory.doSpecialSpawn(entityliving, p_77192_1_, f, l3, f1))
++                                                    if (!net.minecraftforge.event.ForgeEventFactory.doSpecialSpawn(entityliving, p_77192_1_, f, i3, f1))
                                                      ientitylivingdata = entityliving.func_180482_a(p_77192_1_.func_175649_E(new BlockPos(entityliving)), ientitylivingdata);
```
Comment (williewillus) [inline patches/minecraft/net/minecraft/world/WorldEntitySpawner.java.patch]: same here

File: src/main/java/net/minecraftforge/client/model/ModelLoader.java [hunk 3]
```diff
@@ -884,10 +885,10 @@ public boolean hasCustomLoader(IResourceManager manager, ResourceLocation locati
...
         {
-            BufferedImage image = new BufferedImage(16, 16, BufferedImage.TYPE_INT_ARGB);
+            BufferedImage image = new BufferedImage(this.getIconWidth(), this.getIconHeight(), BufferedImage.TYPE_INT_ARGB);
             Graphics2D graphics = image.createGraphics();
             graphics.setBackground(Color.WHITE);
-            graphics.clearRect(0, 0, 16, 16);
+            graphics.clearRect(0, 0, this.getIconWidth(), this.getIconHeight());
             int[][] pixels = new int[Minecraft.getMinecraft().gameSettings.mipmapLevels + 1][];
             pixels[0] = new int[image.getWidth() * image.getHeight()];
             image.getRGB(0, 0, image.getWidth(), image.getHeight(), pixels[0], 0, image.getWidth());
```
Comment (williewillus) [inline src/main/java/net/minecraftforge/client/model/ModelLoader.java]: same here

**Comment by williewillus** [comment]
I'm not in favor of this. It's just one extra call that the modder can make themselves

**Comment by williewillus** [comment]
I also don't like that it's nullable. 
Since pos is never null just return the world state at pos always.
The less that's Nullable the better.

**Comment by cpw** [comment]
This PR is a mess. Reopen it when it's against the main (1.10.x) branch. Also, you need a valid email address in your commit to be able to contribute to MinecraftForge, so you can sign the CLA.

**Pull request closed without merging**